### PR TITLE
(SIMP-5760) Refactor into testable Ruby classes

### PR DESCRIPTION
**Pull request by op-ct**
This patch refactors the `simp_packer_test.sh` and `simp_config.rb` scripts
into testable Ruby classes, and adds a few example unit tests.

* Added:
  - Initial Ruby Unit tests
  - New rake task: `simp:packer:build`
  - New Rake task: `packer:validate`
  - Each matrix log now includes information about its iteration

* Changed:
  - Fixed `VBoxManage hostonlyif ipconfig` logic
  - Refactored `simp_config.rb` and `simp_packer_test.sh` into code under `lib/`
  - Matrix `encryption=on` automatically adds 2 minutes to `big_sleep`

* Removed:
  - Scripts: `simp_packer_test.sh` and `simp_config.rb`

SIMP-5760 #close
SIMP-5238 #comment Refactored scripts into testable classes

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,86 +82,43 @@ Requirements:
...
 same time. A rake task, `simp:packer:matrix` has been provided for this purpose:
 
 ```sh
-[ENV_VAR=value ...] bundle exec rake simp:packer:matrix[MATRIX_ELEMENTS ...]
+TMPDIR=/some/tmp/dir \
+  [ENV_VAR=value ...] \
+  bundle exec rake simp:packer:matrix[MATRIX_ELEMENTS ...]
 ```
 
 A typical example would look like:
```
Comment (lnemsick-simp) [inline README.md]: The discussion about TMPDIR needs to be in this section as well.  See lines 180-181 in README.md in the PR.

File: README.md [hunk 2]
```diff
@@ -204,7 +163,56 @@ Matrix elements are delimited by colons (`:`)
...
-| `MATRIX_LABEL`        | Label for this matrix run that will prefix each iteration's directory name (default: `build`) |
+| `MATRIX_LABEL`        | Label for this matrix run that will prefix each iteration's directory name (default: `build_<YYYYmmdd>_<HHMMSS>`) |
+
+
+### Using `rake simp:packer:build`
+
+```sh
+TMPDIR=/some/tmp/dir \
+  [ENV_VAR=value ...] \
+  bundle exec rake simp:packer:build[vars_json,packer_yaml,simp_conf_yaml,test_dir]
```
Comment (lnemsick-simp) [inline README.md]: What are the arguments to this build task?  Do all need to be specified?  Previously only the directory needed to be specified.

File: README.md [hunk 3]
```diff
@@ -452,17 +444,48 @@ To see a list of development-related tasks available for this project, run:
...
+
+simp-packer is intended to create a family of different SIMP box configurations
+to support automated integration tests for the entire [SIMP Pre-Release
+checklist][simp-prerel-checklist] _(requires [SIMP-5238][SIMP-5238])_:
+
+![simp-packer roadmap](assets/simp-packer-box-roadmap.png)
+
+The following family of box roles can be used to automate integration tests for
+an ISO-based SIMP release:
+
```
Comment (lnemsick-simp) [inline README.md]: Excellent list of box types.  Do we have tickets for each type?

File: README.md (hunk 2 above)
Comment (jeannegreulich) [reply README.md]: I fixed the README

**Pull request merged**